### ISSUE TITLE
NodeGoat docker-compose build fails

### ISSUE THREAD

**Issue by juliadotter** [OWASP/NodeGoat#77]
Hi 

I get this error message when I follow the instructions of OPTION 3 - Run NodeGoat on Docker: 

$sudo docker-compose build
...
npm info linkStuff owasp-nodejs-goat@1.3.0
npm info install owasp-nodejs-goat@1.3.0
npm info postinstall owasp-nodejs-goat@1.3.0
npm info prepublish owasp-nodejs-goat@1.3.0
npm ERR! Linux 4.4.0-31-generic
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "install"
npm ERR! node v4.4.7
npm ERR! npm  v2.15.8
npm ERR! code EPEERINVALID

npm ERR! peerinvalid The package grunt@1.0.1 does not satisfy its siblings' peerDependencies requirements!

I realise it's a problem with Docker/npm/Grunt and not NodeGoat itself, but the effect is that I can't use the NodeGoat docker image, so if there were some known tips and tricks, perhaps you can include them in the instructions.

Thanks

**Comment by binarymist** [comment]
Hi @juliadotter 

I was actually fixing this and about to submit the ticket and realised you had already done it. The below are my findings:

`grunt-if` and `grunt-npm-install` currently have `peerDependencies` and `dependencies` on old versions of grunt (~0.4.5 and ~0.4.1 respectively).
NodeGoat was updated recently to use grunt ^1.0.1.
`grunt-if` and `grunt-npm-instal`l still function with grunt 1.0.1, so the testing of 5127d0ba489161b69710d04fe16648941de60e66 worked fine, but produced the warning `UNMET PEER DEPENDENCY grunt@1.0.1`. My thinking at the time was that this would be fine and that the `grunt-if` and `grunt-npm-install` would get updated to depend on the later version of grunt at some stage.

We encounter a problem when creating our docker images. When the `npm install` is run, the last part of the output is... Pretty much as @juliadotter has mentioned:

```
npm ERR! Linux 3.13.0-24-generic
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "install"
npm ERR! node v4.4.7
npm ERR! npm  v2.15.8
npm ERR! code EPEERINVALID

npm ERR! peerinvalid The package grunt@1.0.1 does not satisfy its siblings' peerDependencies requirements!

npm ERR! Please include the following file with any support request:
npm ERR!     /usr/src/app/npm-debug.log
ERROR: Service 'web' failed to build: The command '/bin/sh -c npm install' returned a non-zero code: 1
```

I'm temporarily downgrading grunt so that the docker-compose build succeeds. As soon as https://github.com/tylerbeck/grunt-if/pull/3 and https://github.com/iclanzan/grunt-npm-install/pull/13 have been merged and published to npm, the grunt version can be updated again to ^1.0.1

Please leave this ticket open until the grunt packages have complied.

**Comment by binarymist** [comment]
If they don't comply which is looking likely, then we have two obvious options:

1. Fork the grunt packages and publish upgraded versions to npm
2. Consider migrating away from gulp

Maybe others?

**Comment by binarymist** [comment]
We've got movement on grunt-npm-install (https://github.com/iclanzan/grunt-npm-install/pull/13#issuecomment-247814748)

**Comment by jesusprubio** [comment]
Recently I've changed all my project to use npm scripts directly. If I need something bigger I add a folder (ie: scripts) with the .js files implemented what I need.

What do you think? I could do it quickly. (/cc: @ckarande)

Ref: https://medium.freecodecamp.com/why-i-left-gulp-and-grunt-for-npm-scripts-3d6853dd22b8#.n8yb9qu9g

**Comment by binarymist** [comment]
It's just `grunt-if` left.
What's more work out of:

1. Fork grunt-if, update the peerDependency, publish to NPM, update NodeGoat to use new grunt package and update the grunt version
2. Write a script to replace the work that grunt-if already does, include this new script, replace how we are using grunt-if / modify the grunt file.

Is writing our own logic a better option than consuming someone elses? Usually I'd say no, but maybe it is in this case, as we don't have to depend on an unengaged maintainer and the amount of logic we would need to provide would be trivial. This is sounding like a good idea @jesusprubio . Thoughts @ckarande ?

A note on that link @jesusprubio There are so many flawed arguments in that link it's not funny. NPM scripts definitely have their place, but that article has many unfounded statements, so not a great article for selling NPM scripts.

**Comment by jesusprubio** [comment]
I have started a little refactor to move to npm-scripts but I have a couple of doubts:
- Why are we using nodemon and grunt-watch? Both are used for the same thing. Moreover they're pointing to the same folders, so it's a bit strange for me.
- Why do we have a task to beautify .js stuff?

Thanks! :)

**Comment by binarymist** [comment]
Fixed in #86
